fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/bitmap/ComposedBitmapIndexQueryPlan.java

```diff
@@ -0,0 +1,496 @@
+import com.apple.foundationdb.record.query.plan.temp.CorrelationIdentifier;
+import com.apple.foundationdb.record.query.plan.temp.RelationalExpression;
+import com.apple.foundationdb.record.query.plan.temp.explain.NodeInfo;
+import com.apple.foundationdb.record.query.plan.temp.explain.PlannerGraph;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A query plan implementing a bit-wise merge of two or more covering index scans of {@code BITMAP_VALUE} indexes.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class ComposedBitmapIndexQueryPlan implements RecordQueryPlanWithNoChildren {
+
+    @Nonnull
+    // NOTE: These aren't children in the sense of RecordQueryPlanWithChildren
+    // for the same reason as RecordQueryCoveringIndexPlan isn't RecordQueryPlanWithChild.
+    private final List<RecordQueryCoveringIndexPlan> indexPlans;
+    @Nonnull
+    private final ComposerBase composer;
+
+    ComposedBitmapIndexQueryPlan(@Nonnull List<RecordQueryCoveringIndexPlan> indexPlans, @Nonnull ComposerBase composer) {
+        this.indexPlans = indexPlans;
+        this.composer = composer;
+    }
+
+    @Nonnull
+    public List<RecordQueryCoveringIndexPlan> getIndexPlans() {
+        return indexPlans;
+    }
+
+    @Nonnull
+    public ComposerBase getComposer() {
+        return composer;
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("unchecked")
+    public <M extends Message> RecordCursor<FDBQueriedRecord<M>> execute(@Nonnull FDBRecordStoreBase<M> store, @Nonnull EvaluationContext context,
+                                                                         @Nullable byte[] continuation, @Nonnull ExecuteProperties executeProperties) {
+        final ExecuteProperties scanExecuteProperties = executeProperties.getSkip() > 0 ? executeProperties.clearSkipAndAdjustLimit() : executeProperties;
+        final List<Function<byte[], RecordCursor<IndexEntry>>> cursorFunctions = indexPlans.stream()
+                .map(RecordQueryCoveringIndexPlan::getIndexPlan)
+                .map(scan -> (Function<byte[], RecordCursor<IndexEntry>>) childContinuation -> scan.executeEntries(store, context, childContinuation, scanExecuteProperties))
+                .collect(Collectors.toList());
+        return ComposedBitmapIndexCursor.create(cursorFunctions, composer, continuation, store.getTimer())
+                // Composers can return null bitmaps when empty, which is then left out of the result set.
+                .filter(indexEntry -> indexEntry.getValue().get(0) != null)
+                .map(indexPlans.get(0).indexEntryToQueriedRecord(store));
+    }
+
+    @Override
+    public boolean isReverse() {
+        return false;
```

Comment:
Yes, I think it's kind of ill-defined. Like does the whole bitmap reverse or just the enumeration order?

Reply:
Yeah. I...think it would reverse the whole bitmap, though I suppose in some sense, that's a little Endian vs big Endian debate. In any case, leaving it as not allowed (at least for now) rather than trying to settle that debate seems like the right call to me.